builder/builder.go

```diff
@@ -203,3 +207,76 @@ func Receipt(fee *felt.Felt, feeUnit core.FeeUnit, txHash *felt.Felt, trace *vm.
 		RevertReason:       trace.RevertReason(),
 	}
 }
+
+func (b *Builder) listenPool(ctx context.Context) error {
+	for {
+		if err := b.depletePool(ctx); err != nil {
+			if !errors.Is(err, db.ErrKeyNotFound) {
+				return err
+			}
+		}
+
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-b.pool.Wait():
+			continue
+		}
```

Comment:
What do you think about having three cases?

1. `ctx.Done()` -> exit
2. `b.pool.Wait()` -> add tx to the pending block. If after adding the transaction, we reach a certain `maxTransactions` threshold, commit the block.
3. `time.After` -> timeout, commit the block.

Reply:
Sadly, that doesnt really work without nesting two for loops.
Mainly because `b.pool.Wait()` can block even with txns in the pool. Wait() is only supposed to be called when we consume all the txns in the pool.